graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java

```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
     }
 
     private boolean hasInvalidInputParamType(ParameterSymbol inputTypeSymbol) {
-        return !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
+        boolean hasInvalidParamType = true;
+        if (inputTypeSymbol.typeDescriptor().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            if ((((TypeReferenceTypeSymbol) inputTypeSymbol.typeDescriptor()).definition()).kind() == SymbolKind.ENUM) {
+                hasInvalidParamType = false;
```

Comment:
can't we return directly here?
```suggestion
                return false;
```